os/kernel/sched/sched_releasetcb.c

```diff
@@ -211,6 +214,22 @@ int sched_releasetcb(FAR struct tcb_s *tcb, uint8_t ttype)
 
 		ret = up_addrenv_detach(tcb->group, tcb);
 #endif
+#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(CONFIG_ARM_MPU)
+		/* Disable mpu regions when binary is unloaded */
+		if (IS_LOADED_MODULE(tcb->group)) {
```

Comment:
How about using the conditionals like below?
```c
if (IS_LOADED_MODULE(tcb->group) && up_mpu_disable_region(&tcb->mpu_regs[0]) {
```
I think there is no case that 0 index is the same but 1 and 2 index is different. So just checking mpu_regs[0] is enough.
Then we can reduce duplicated code as above.